### PR TITLE
feat(number-input): add new stepper events and props Al 5312

### PR DESCRIPTION
**Pull request by ianaa**
I wanted to also update to the docs to follow the new storybook format, but opening this PR as is because I'm sick again 😞

### CODE REVIEW

File: src/stories/NumberInput.mdx [hunk 1]
```diff
@@ -84,11 +84,22 @@ To provide a custom placeholder, use the `placeholder` prop.
...
+<Canvas>
   <Story id="components-numberinput--step-up"></Story>
 </Canvas>
+
+## Listening to value updates
+
+There are several events available which indicate that the value has changed:
+
+- `@update:model-value` is emitted each time the model value updates,
+  both when the value is inputted directly, or changed with input stepping functions
```
Comment (manuatarchilogic) [inline src/stories/NumberInput.mdx:103]: TIL both "inputted" and "input" are accepted as past participle (I thought "inputted" was wrong)

File: src/components/__tests__/NumberInput.test.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import { render } from '@testing-library/vue'
+import userEvent from '@testing-library/user-event'
+
+import NumberInput from '../NumberInput.vue'
+
+describe('NumberInput.vue', () => {
+  describe('when the input value is changed by typing in new value', () => {
```
Comment (manuatarchilogic) [inline src/components/__tests__/NumberInput.test.ts]:
```suggestion
  describe('when the input value is changed by typing a new value', () => {
```

File: src/components/__tests__/NumberInput.test.ts
Comment (manuatarchilogic) [inline src/components/__tests__/NumberInput.test.ts:1]: nice tests!

**Review by manuatarchilogic — COMMENTED** [review]
I would have expected this to go: 10, 11, 12, 22, 32, 42 …
not: 10, 11, 12, 20, 30, 40 …


https://github.com/archilogic-com/honeycomb/assets/89903103/49919d87-92f0-4f5b-a269-d455ee3ed2ae

**Comment by ianaa** [comment]
> I would have expected this to go: 10, 11, 12, 22, 32, 42 … not: 10, 11, 12, 20, 30, 40 …
> 

@manuatarchilogic me too, but that's just how the steps work natively for some reason

https://github.com/archilogic-com/honeycomb/assets/22019313/b37da6a1-4454-47c2-ac22-dd29bb01f810

**Review by manuatarchilogic — APPROVED** [review]

**Pull request merged**